File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    clothesMap = {}
    for item, type in clothes:
        if type not in clothesMap:
            clothesMap[type] = 0
        clothesMap[type] += 1
    
    combinations = 1
    for val in clothesMap.values():
        combinations *= (val + 1)
    
    return combinations-1
        
        

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const clothMap = {};
    for(const [item, type] of clothes){
        if(!clothMap[type]) clothMap[type] = 0;
        clothMap[type] += 1;
    }
    
    let combinations = 1;
    for(const type in clothMap){
        combinations *= (clothMap[type] + 1);
    }
    
    return combinations - 1;
}